public/app/features/alerting/unified/components/amroutes/AmRoutesExpandedForm.tsx

```diff
@@ -0,0 +1,219 @@
+import React, { FC, useState } from 'react';
+import { css } from '@emotion/css';
+import { GrafanaTheme, SelectableValue } from '@grafana/data';
+import {
+  Button,
+  Checkbox,
+  Field,
+  FieldArray,
+  Form,
+  HorizontalGroup,
+  Input,
+  InputControl,
+  MultiSelect,
+  Select,
+  Switch,
+  useStyles,
+} from '@grafana/ui';
+import { Receiver, Route } from 'app/plugins/datasource/alertmanager/types';
+import { parseInterval, timeOptions } from '../../utils/time';
+import { AmRoutesTable } from './AmRoutesTable';
+
+export interface AmRoutesExpandedReadProps {
+  onExitEditMode: () => void;
+  route: Route;
+  receivers: Array<SelectableValue<Receiver['name']>>;
+}
+
+const prepareMatches = (matches: Record<string, string> | undefined, isRegex: boolean) =>
+  Object.entries(matches ?? {}).reduce(
+    (acc, [label, value]) => [
+      ...acc,
+      {
+        label,
+        value,
+        isRegex,
+      },
+    ],
+    []
+  );
+
+export const AmRoutesExpandedForm: FC<AmRoutesExpandedReadProps> = ({ onExitEditMode, route, receivers }) => {
+  const styles = useStyles(getStyles);
+  const [overrideGrouping, setOverrideGrouping] = useState(!!route.group_by);
+  const [overrideTimings, setOverrideTimings] = useState(
+    !!route.group_wait || !!route.group_interval || !!route.repeat_interval
+  );
+
+  const [groupByOptions, setGroupByOptions] = useState(
+    (route.group_by ?? []).map((opt) => ({
+      label: opt,
+      value: opt,
+    }))
+  );
+
+  const [groupWaitValue, groupWaitValueType] = route?.group_wait
+    ? parseInterval(route?.group_wait)
+    : [undefined, undefined];
+
+  const [groupIntervalValue, groupIntervalValueType] = route?.group_interval
+    ? parseInterval(route?.group_interval)
+    : [undefined, undefined];
+
+  const [repeatIntervalValue, repeatIntervalValueType] = route?.repeat_interval
+    ? parseInterval(route?.repeat_interval)
+    : [undefined, undefined];
+
+  const defaultValues = {
```

Comment:
I would maybe wrap calculation of `defaultValues` in `useMemo`. But don't have solid arguments or feel strongly about it

Reply:
I'm still trying to figure out if `useMemo` makes sense.

Usually it would because the `Form` prop would just listen to `defaultValues` and reset if you pass another set of default values. But! It doesn't seem the case here. I'm still looking into it.

Reply:
yeah, I'd do it to not worry about such issues, but otherwise don't have strong arguments for it :)